traffic_ops/app/lib/API/Configs/ApacheTrafficServer.pm

```diff
@@ -0,0 +1,1959 @@
+	elsif ( $filename eq "remap.config" ) { $file_contents = $self->remap_dot_config( $server_obj, $filename, $scope ); }
+	elsif ( $filename eq "hosting.config" ) { $file_contents = $self->hosting_dot_config( $server_obj, $filename, $scope ); }
+	elsif ( $filename eq "cache.config" ) { $file_contents = $self->cache_dot_config( $server_obj, $filename, $scope ); }
+	elsif ( $filename eq "packages" ) {
+		$file_contents = $self->get_package_versions( $server_obj, $filename, $scope );
+		$file_contents = encode_json($file_contents);
+	}
+	elsif ( $filename eq "chkconfig" ) {
+		$file_contents = $self->get_chkconfig( $server_obj, $filename, $scope );
+		$file_contents = encode_json($file_contents);
+	}
+	else {
+		my $file_param = $self->db->resultset('Parameter')->search( [ config_file => $filename ] )->single;
+		if ( !defined($file_param) ) {
+			return $self->not_found();
+		}
+		$file_contents = $self->take_and_bake_server( $server_obj, $filename, $scope );
+	}
+
+	#if we get an empty file, just send back an error.
+	if ( !defined($file_contents) ) {
+		return $self->not_found();
+	}
+
+	#return the file contents for fetch and db actions.
+	return $self->render( text => $file_contents, format => 'txt' );
+}
+
+#entry point for cdn scope api route.
+sub get_cdn_config {
+	my $self     = shift;
+	my $filename = $self->param("filename");
+	my $id       = $self->param('id');
+	my $scope    = $self->get_scope($filename);
+
+	##check user access
+	if ( !&is_oper($self) ) {
+		return $self->forbidden();
+	}
+	print STDERR "well i'm here anyway.";
+	##check the scope - is this the correct route?
+	if ( $scope ne 'cdn' ) {
+		return $self->alert( "Error - incorrect file scope for route used.  Please use the " . $scope . " route." );
+	}
+
+	##verify that a valid cdn ID has been used
+	my $cdn_obj = $self->cdn_data($id);
+	if ( !defined($cdn_obj) ) {
+		return $self->not_found();
+	}
+
+	#generate the config file using the appropriate function
+	my $file_contents;
+	if ( $filename eq "bg_fetch.config" ) { $file_contents = $self->bg_fetch_dot_config( $cdn_obj, $filename, $scope ); }
+	elsif ( $filename =~ /cacheurl.*\.config/ ) { $file_contents = $self->cacheurl_dot_config( $cdn_obj, $filename, $scope ); }
+	elsif ( $filename =~ /hdr_rw_.*\.config/ ) { $file_contents = $self->header_rewrite_dot_config( $cdn_obj, $filename, $scope ); }
+	elsif ( $filename =~ /regex_remap_.*\.config/ ) { $file_contents = $self->regex_remap_dot_config( $cdn_obj, $filename, $scope ); }
+	elsif ( $filename eq "regex_revalidate.config" ) { $file_contents = $self->regex_revalidate_dot_config( $cdn_obj, $filename, $scope ); }
+	elsif ( $filename =~ /set_dscp_.*\.config/ ) { $file_contents = $self->set_dscp_dot_config( $cdn_obj, $filename, $scope ); }
+	elsif ( $filename eq "ssl_multicert.config" ) { $file_contents = $self->ssl_multicert_dot_config( $cdn_obj, $filename, $scope ); }
+	else                                          { return $self->not_found(); }
```

Comment:
You can probably take the "else" out and just let it get caught in the 

if ( !defined($file_contents) ) {
  return $self->not_found();
 }